pkg/operator/controller.go

```diff
@@ -420,39 +478,159 @@ func (c *Controller) syncProfile(tuned *tunedv1.Tuned, nodeName string) error {
 				return err
 			}
 
-			klog.V(2).Infof("Profile %s not found, creating one", profileMf.Name)
+			klog.V(2).Infof("syncProfile(): Profile %s not found, creating one [%s]", profileMf.Name, tunedProfileName)
 			profileMf.Spec.Config.TunedProfile = tunedProfileName
 			_, err = c.clients.Tuned.TunedV1().Profiles(ntoconfig.OperatorNamespace()).Create(profileMf)
-
 			if err != nil {
-				return fmt.Errorf("failed to create Profile: %v", err)
+				return fmt.Errorf("failed to create Profile %s: %v", profileMf.Name, err)
 			}
-			// DaemonSet created successfully
+			// Profile created successfully
+			klog.Infof("created profile %s [%s]", profileMf.Name, tunedProfileName)
 			return nil
 		}
 
 		return fmt.Errorf("failed to get Profile %s: %v", profileMf.Name, err)
-	} else {
-		if profile.Spec.Config.TunedProfile == tunedProfileName {
-			klog.V(2).Infof("no need to update Profile %s", nodeName)
+	}
+
+	if mcLabels != nil {
+		// The tuned profile "tunedProfileName" for nodeName matched with MachineConfig
+		// labels set for additional machine configuration.  Sync the operator-created
+		// MachineConfig for MachineConfigPools 'pools'.
+		err := c.syncMachineConfig(getMachineConfigNameForPools(pools), mcLabels, profile.Status.Bootcmdline)
```

Comment:
Is  `profile.Status.Bootcmdline` Represents all tuned kernel arguments that should be added by the mc ? Does it include all argument for the base profiles in use ? (e.g. realtime + network latency arguments)

Reply:
Yes, it does.